src/docfx/build/apply-templates/ApplyTemplates.cs

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Docs.Build
+{
+    internal static class ApplyTemplates
+    {
+        public static bool Run(ErrorBuilder errors, CommandLineOptions options)
+        {
+            if (string.IsNullOrEmpty(options.Template))
+            {
+                throw new InvalidOperationException("Must provide the path to template repo when apply templates.");
+            }
+
+            var inputDir = options.WorkingDirectory;
+            var outputDir = options.Output ?? Path.Combine(AppContext.BaseDirectory, "outputs");
+            var package = new LocalPackage(options.Template);
+            var locale = options.Locale ?? "en-us";
+            var trustedObj = JToken.Parse(File.ReadAllText(Path.Combine(AppContext.BaseDirectory, "data/docs/trusted-domains.json")));
+            var trustedDomains = JsonUtility.ToObject<Dictionary<string, TrustedDomains>>(errors, trustedObj);
+            var config = new Config
+            {
+                TrustedDomains = trustedDomains,
+                DryRun = false,
+                Template = new PackagePath(options.Template),
+            };
+
+            var fileResolver = new FileResolver(package);
+            var jsonSchemaLoader = new JsonSchemaLoader(fileResolver);
+            var jsonSchemaProvider = new JsonSchemaProvider(config, package, jsonSchemaLoader);
+            var templateEngine = TemplateEngine.CreateTemplateEngine(errors, config, null, locale, package, fullBuild: false);
+
+            Parallel.ForEach(
```

Comment:
Use `ParallelUtility.ForEach`, it has logic to handle exceptions.

Reply:
Done